### PR TITLE
Sstate signatures

### PR DESCRIPTION
**Pull request by shr-project**

### CODE REVIEW

**Comment by shr-project** [comment]
@Tofee @Herrie82 FYI

File: recipes-graphics/mesa/mesa_%.bbappend [hunk 1]
```diff
@@ -1,4 +1,8 @@
 # to debug some issues
 # EXTRA_OEMESON:append:pinephone = " --buildtype=debug "
 
-PACKAGECONFIG:append:pinephone = " kmsro lima"
+# enable both lima and panfrost to be able to reuse mesa sstate between pp and ppp
+# meta-rockchip is enabling panfrost for rk3399:
+# mesa_%.bbappend:PACKAGECONFIG:append:rk3399 = " kmsro panfrost"
+PACKAGECONFIG:append:pinephone = " kmsro lima panfrost"
+PACKAGECONFIG:append:pinephonepro = " kmsro lima panfrost"
```
Comment (Herrie82) [inline recipes-graphics/mesa/mesa_%.bbappend:8]: @shr-project Looks good to me.

Comment (Tofee) [reply recipes-graphics/mesa/mesa_%.bbappend:8]: Same

**Comment by shr-project** [comment]
Remaining issues:
```
webos-ports $ bitbake-diffsigs tmp-glibc/sstate-diff/1642170508/p*/*/luna-next/*do_configure.sig*
NOTE: Starting bitbake server...
Hash for dependent task luna-next/luna-next.bb:do_generate_toolchain_file changed from 994c4c0108fe83b2d942a160a054db2f60792a71f2667d8f39fcd35d7e9ccfc5 to f4305b65d0bb8983f4fc1c18bd8dfa0d7ccb327d50a3e58ad36b983de2847b36
Unable to find matching sigdata for /OE/build/luneos-honister/webos-ports/meta-webos-ports/meta-luneui/recipes-luneui/luna-next/luna-next.bb:do_generate_toolchain_file with hashes 994c4c0108fe83b2d942a160a054db2f60792a71f2667d8f39fcd35d7e9ccfc5 or f4305b65d0bb8983f4fc1c18bd8dfa0d7ccb327d50a3e5
8ad36b983de2847b36
OE qemux86-64@luneos /OE/build/luneos-honister/webos-ports $ bitbake-diffsigs tmp-glibc/sstate-diff/1642170508/p*/*/luna-next/*do_gen*
NOTE: Starting bitbake server...
basehash changed from 354b6b84a02628cf5688cbf0147197f11f657775daa463081410ec376a5d7217 to c497738d558d48a40b2058142c6a466b9949608af102bc80cfdc51f7d17f7270
Variable CXXFLAGS value changed:
"${TARGET_CXXFLAGS} -fvisibility-inlines-hidden  -DMESA_EGL_NO_X11_HEADERS=1   -DQT_WAYLAND_COMPOSITOR_QUICK  ${HOST_CC_ARCH} ${TOOLCHAIN_OPTIONS} [--DEGL_NO_X11=1 -]"
```

```
webos-ports $ bitbake-diffsigs tmp-glibc/sstate-diff/1642170508/p*/*/trusted-firmware-a/*do_fetch*
NOTE: Starting bitbake server...
basehash changed from f025a9e278744e10e02d41a912092d06d112f9aed7b1b3860aa6beab6fa36fd7 to caa1a45aa9703bbd1531c7df26ff77c29a1567a93690d025e64e826c3e58197e
Variable SRC_URI value changed:
"git://git.trustedfirmware.org/TF-A/trusted-firmware-a.git;protocol=https;branch=${BRANCH}     file://0001-dram-Fix-build-with-gcc-11.patch     file://0001-plat_macros.S-Use-compatible-.asciz-asm-directive.patch     file://0001-pmu-Do-not-mark-already-defined-functions-as-weak.patch  {+    fil
e://0001-rk3399-baudrate.patch     +}"
Dependency on checksum of file 0001-rk3399-baudrate.patch was added
```

```webos-ports $ bitbake-diffsigs tmp-glibc/sstate-diff/1642170508/p*/*/linux-firmware-pine64/*do_install.sig*
NOTE: Starting bitbake server...
Task dependencies changed from:
['ABIEXTENSION', 'ABIEXTENSION_32', 'ABIEXTENSION_64', 'AR', 'ARMPKGSFX_ENDIAN_64', 'AS', 'ASNEEDED', 'BUILD_AR', 'BUILD_AS', 'BUILD_AS_ARCH', 'BUILD_CC', 'BUILD_CCLD', 'BUILD_CC_ARCH', 'BUILD_CFLAGS', 'BUILD_CPP', 'BUILD_CPPFLAGS', 'BUILD_CXX', 'BUILD_CXXFLAGS', 'BUILD_FC', 'BUILD_LD', 'BUILD
_LDFLAGS', 'BUILD_LD_ARCH', 'BUILD_NM', 'BUILD_OPTIMIZATION', 'BUILD_PREFIX', 'BUILD_RANLIB', 'BUILD_STRIP', 'CC', 'CCLD', 'CFLAGS', 'CPP', 'CPPFLAGS', 'CXX', 'CXXFLAGS', 'D', 'DEBUG_BUILD', 'DEBUG_FLAGS', 'DEBUG_OPTIMIZATION', 'DEBUG_PREFIX_MAP', 'EXTENDPE', 'FC', 'FULL_OPTIMIZATION', 'HOST_A
S_ARCH', 'HOST_CC_ARCH', 'HOST_LD_ARCH', 'HOST_PREFIX', 'LC_ALL', 'LD', 'LDFLAGS', 'LIBCEXTENSION', 'LINKER_HASH_STYLE', 'MAKE', 'NM', 'OBJCOPY', 'OBJDUMP', 'PE', 'PERL_HASH_SEED', 'PKG_CONFIG_DIR', 'PKG_CONFIG_DISABLE_UNINSTALLED', 'PKG_CONFIG_LIBDIR', 'PKG_CONFIG_PATH', 'PKG_CONFIG_SYSROOT_D
IR', 'PKG_CONFIG_SYSTEM_INCLUDE_PATH', 'PKG_CONFIG_SYSTEM_LIBRARY_PATH', 'PN', 'PR', 'PSEUDO_DISABLED', 'PV', 'PYTHONHASHSEED', 'RANLIB', 'READELF', 'RECIPE_SYSROOT_NATIVE', 'REMOVE_LIBTOOL_LA', 'SELECTED_OPTIMIZATION', 'SOURCE_DATE_EPOCH_FALLBACK', 'STAGING_BASE_LIBDIR_NATIVE', 'STAGING_DATAD
IR', 'STAGING_DIR_NATIVE', 'STAGING_INCDIR_NATIVE', 'STAGING_LIBDIR_NATIVE', 'STRINGS', 'STRIP', 'TARGET_ARCH', 'TARGET_AS_ARCH', 'TARGET_CC_ARCH', 'TARGET_CFLAGS', 'TARGET_CPPFLAGS', 'TARGET_CXXFLAGS', 'TARGET_LDFLAGS', 'TARGET_LD_ARCH', 'TARGET_LINK_HASH_STYLE', 'TARGET_OS', 'TARGET_PREFIX',
 'TARGET_SYS', 'TARGET_VENDOR', 'TOOLCHAIN_OPTIONS', 'TUNE_ARCH', 'TUNE_ARCH_32', 'TUNE_ARCH_64', 'TUNE_ASARGS', 'TUNE_CCARGS', 'TUNE_LDARGS', 'TZ', 'base_bindir', 'base_libdir', 'base_libdir_native', 'base_prefix', 'base_sbindir', 'baselib', 'bindir', 'datadir', 'do_install[fakeroot]', 'docdi
r', 'exec_prefix', 'includedir', 'includedir_native', 'infodir', 'libdir', 'libdir_native', 'libexecdir', 'localstatedir', 'mandir', 'nonarch_base_libdir', 'nonarch_libdir', 'oldincludedir', 'prefix', 'prefix_native', 'remove_libtool_la', 'root_prefix', 'sbindir', 'servicedir', 'sharedstatedir
', 'sysconfdir', 'systemd_system_unitdir', 'systemd_unitdir', 'systemd_user_unitdir']
to:
['ABIEXTENSION', 'ABIEXTENSION_32', 'ABIEXTENSION_64', 'AR', 'ARMPKGSFX_ENDIAN_64', 'AS', 'ASNEEDED', 'BUILD_AR', 'BUILD_AS', 'BUILD_AS_ARCH', 'BUILD_CC', 'BUILD_CCLD', 'BUILD_CC_ARCH', 'BUILD_CFLAGS', 'BUILD_CPP', 'BUILD_CPPFLAGS', 'BUILD_CXX', 'BUILD_CXXFLAGS', 'BUILD_FC', 'BUILD_LD', 'BUILD
_LDFLAGS', 'BUILD_LD_ARCH', 'BUILD_NM', 'BUILD_OPTIMIZATION', 'BUILD_PREFIX', 'BUILD_RANLIB', 'BUILD_STRIP', 'CC', 'CCLD', 'CFLAGS', 'CPP', 'CPPFLAGS', 'CXX', 'CXXFLAGS', 'D', 'DEBUG_BUILD', 'DEBUG_FLAGS', 'DEBUG_OPTIMIZATION', 'DEBUG_PREFIX_MAP', 'EXTENDPE', 'FC', 'FULL_OPTIMIZATION', 'HOST_A
S_ARCH', 'HOST_CC_ARCH', 'HOST_LD_ARCH', 'HOST_PREFIX', 'LC_ALL', 'LD', 'LDFLAGS', 'LIBCEXTENSION', 'LINKER_HASH_STYLE', 'MAKE', 'NM', 'OBJCOPY', 'OBJDUMP', 'PE', 'PERL_HASH_SEED', 'PKG_CONFIG_DIR', 'PKG_CONFIG_DISABLE_UNINSTALLED', 'PKG_CONFIG_LIBDIR', 'PKG_CONFIG_PATH', 'PKG_CONFIG_SYSROOT_D
IR', 'PKG_CONFIG_SYSTEM_INCLUDE_PATH', 'PKG_CONFIG_SYSTEM_LIBRARY_PATH', 'PN', 'PR', 'PSEUDO_DISABLED', 'PV', 'PYTHONHASHSEED', 'RANLIB', 'READELF', 'RECIPE_SYSROOT_NATIVE', 'REMOVE_LIBTOOL_LA', 'S', 'SELECTED_OPTIMIZATION', 'SOURCE_DATE_EPOCH_FALLBACK', 'STAGING_BASE_LIBDIR_NATIVE', 'STAGING_
DATADIR', 'STAGING_DIR_NATIVE', 'STAGING_INCDIR_NATIVE', 'STAGING_LIBDIR_NATIVE', 'STRINGS', 'STRIP', 'TARGET_ARCH', 'TARGET_AS_ARCH', 'TARGET_CC_ARCH', 'TARGET_CFLAGS', 'TARGET_CPPFLAGS', 'TARGET_CXXFLAGS', 'TARGET_LDFLAGS', 'TARGET_LD_ARCH', 'TARGET_LINK_HASH_STYLE', 'TARGET_OS', 'TARGET_PRE
FIX', 'TARGET_SYS', 'TARGET_VENDOR', 'TOOLCHAIN_OPTIONS', 'TUNE_ARCH', 'TUNE_ARCH_32', 'TUNE_ARCH_64', 'TUNE_ASARGS', 'TUNE_CCARGS', 'TUNE_LDARGS', 'TZ', 'base_bindir', 'base_libdir', 'base_libdir_native', 'base_prefix', 'base_sbindir', 'baselib', 'bindir', 'datadir', 'do_install[fakeroot]', '
docdir', 'exec_prefix', 'includedir', 'includedir_native', 'infodir', 'libdir', 'libdir_native', 'libexecdir', 'localstatedir', 'mandir', 'nonarch_base_libdir', 'nonarch_libdir', 'oldincludedir', 'prefix', 'prefix_native', 'remove_libtool_la', 'root_prefix', 'sbindir', 'servicedir', 'sharedsta
tedir', 'sysconfdir', 'systemd_system_unitdir', 'systemd_unitdir', 'systemd_user_unitdir']
basehash changed from c2b6493e0d9ff4bd48ccfcecf43f37468feb209550b178fb1eec2bf8b15db1dd to 63abaebeff87ef70d16af28a483d8e4bbe3afb8d73d54c65cfc53705089a5afb
Dependency on variable S was added
Variable do_install value changed:
@@ -6,3 +6,9 @@
     install -m 0644 ${WORKDIR}/rtl8723cs_xx_fw.bin ${D}/lib/firmware/rtl_bt/rtl8723cs_xx_fw.bin
     install -m 0644 ${WORKDIR}/rtl8723cs_xx_config.bin ${D}/lib/firmware/rtl_bt/rtl8723cs_xx_config.bin
     install -m 0644 ${WORKDIR}/git/ov5640cam/ov5640_af.bin ${D}/lib/firmware/ov5640_af.bin
+    install -d ${D}/lib/firmware/rockchip/
+    install -d ${D}/lib/firmware/brcm/
+    install -m 0644 ${WORKDIR}/git/pinerock/rockchip/dptx.bin ${D}/lib/firmware/rockchip/dptx.bin
+    install -m 0644 ${S}/ap6256bt/BCM4345C5.hcd ${D}/lib/firmware/brcm
+    install -m 0644 ${WORKDIR}/git/wifinonfree/brcm/brcmfmac4345* ${D}/lib/firmware/brcm
+    ln -s brcmfmac43456-sdio.txt ${D}/lib/firmware/brcm/brcmfmac43456-sdio.pine64,pinephone-pro.txt
```

**Pull request merged**